### DISCUSSION TITLE
Drone Cat Mouse: New features and improvements

### DISCUSSION THREAD

**Discussion by pariaspe** [JdeRobot/RoboticsAcademy#1246] [Ideas]
I'd like to discuss what new features and improvements should we add to drone exercises, starting with **drone cat mouse** exercise.

I've already thought some like:
- Connection protocol extended to both drones.
- Standardization of mouse levels of difficulty.
- Teleoperated mouse.

Any suggestion or idea is welcome!

**Comment by RUFFY-369** [comment]
Hi @pariaspe , Can we add something like cylindrical obstacles to modify the world which have diameter small enough to not totally block the vision system of the drone. And within this world full of obstacles the cat should chase the mouse.
Though this thing can be added as a supplementary for increasing the difficulty of the exercise according to the student' s need.

**Reply by jmplaza** [reply]
We have recently implemented the on-the-fly selection of Gazebo world for the same exercise (in the FollowLine exercise so far, https://github.com/JdeRobot/RoboticsAcademy/issues/1179). So I guess we could develop new Gazebo worlds for this drone-cat-mouse exercise too :-).

**Reply by RUFFY-369** [reply]
Yeah or may be to more of the drone exercises which may or may not be lacking rich world like 'follow_turtlebot'.

**Comment by shailigandhi-oss** [comment]
Hi @pariaspe , @jmplaza ,Protocol for both drones: Right now the student-facing WebSocket stream is mostly cat-centric (#gui / #gul). Extending the payload with mouse pose (and optionally a mouse camera) under a small versioned JSON shape would make the frontend honest about the full scenario and avoid one-off tags piling up.

Standardized mouse difficulty: Levels today live in obfuscated motion code (path_obfuscated), while bounds already exist in drone_cat_mouse.yaml. Moving difficulty to named presets in YAML (speed caps, pattern type, noise, etc.) would make levels documentable, testable, and teachable without changing the “game” idea.

Happy to help break this into incremental PR-sized pieces (protocol first, then YAML levels, then teleop) if maintainers want to sequence it that way.

**Reply by jmplaza** [reply]
Hi @shailigandhi-oss,

The focus of project #4 is not the protocol for both drones. The focus of this project is about synchronization of two robotics applications in the RAM. Current RAM launches the simulated robot world and ONE robotics application (the one programmed by the RoboticsAcademy --RA-- user). After the project the RAM should launch TWO robotics applications: the user one (cat drone) and another one (mouse drone). For simplicity, the mouse drone source code may be placed at the RA directory for that exercise (it would be a system flle for now). And both applications should start  (and reset when needed) at the same time. Each one is in fact a linux process inside the RADI. So, maybe some kind of signal coordination will be needed (the RAM starting first application, sending a STOP signal to it, starting the second application, sending a STOP signal to it also, and finally sending them a RESUME signal). Other synchronization mechanisms should be explored too. There will be no explicit communication between the two applications.

In addition,  I'm afraid is too late for a sensible feedback from the potential mentors and for meaningful contributions on your side before the submission deadline. Just follow the [applicant instructions](https://jderobot.github.io/activities/gsoc/2026#application-instructions-for-gsoc-2026), fill the required form and take a look at [https://github.com/orgs/JdeRobot/discussions/383](https://github.com/orgs/JdeRobot/discussions/383).
